Terasology/LightAndShadow
# feat: game over screen rework

## Purpose

This PR will hold commits regarding tasks from issue Game Over Screen Overhaul https://github.com/Terasology/LightAndShadow/issues/130.

## How to test

1. start a LAS game
2. score 5 points by taking the enemy flag, and bringing it to your base*
3. the screen shall appear when the last point is scored

*you can score points by clicking the box  on the enemy base. To avoid scoring 5 points, you can go to file `LASUtils.java`, and change `GOAL_SCORE = 5` to `GOAL_SCORE = 1`, then you would only need to score 1 point.

## Result
### By checking and the above `How to test` steps, you will get the following result
![image](https://user-images.githubusercontent.com/48293545/90174449-a9245500-ddae-11ea-91d4-3ce1cc20606d.png)

### MigLayout areas Highlighted

Not necessary for checking, but if you want to get the MigLayout areas highlighted, you can replace the `gameoverScreen.skin`'s code with the following lines:
```
{
    "inherit": "default",
    "elements": {
        "UIScreenLayer": {
            "background-scale-mode": "stretch",
            "background": "engine:greyedBackground"
        },
        "UIButton": {
            "min-width": 128,
            "min-height": 32
        },
        "UILabel": {
            "font" : "lightAndShadowResources:WickedQueen"
        },
        "MigLayout": {
            "background-scale-mode": "stretch",
            "background": "engine:greyedBackground"
        }
    }
}
```

![image](https://user-images.githubusercontent.com/48293545/90174511-ce18c800-ddae-11ea-9325-4fba36e5ee36.png)
